Sources/BSplineLib/ParameterSpaces/knot_vector.cpp

```diff
@@ -185,14 +185,52 @@ KnotVector::DetermineMultiplicity(Knot const& parametric_coordinate,
     Throw(exception, kName);
   }
 #endif
-  return Multiplicity{static_cast<int>(std::count_if(
-      knots_.begin(),
-      knots_.end(),
-      [&](Knot const& current_knot) {
-        return std::abs(current_knot - parametric_coordinate) < tolerance;
-      }))};
+  auto knot_iter = knots_.cbegin();
+
+  // initialize multiplicity and loop around
+  int multiplicity{};
+  for (; knot_iter != knots_.cend();) {
+    if (std::abs(parametric_coordinate - *knot_iter++) < tolerance) {
+      ++multiplicity;
+      while (std::abs(parametric_coordinate - *knot_iter++) < tolerance) {
+        ++multiplicity;
+      }
+      break;
+    }
+  }
+
+  return Multiplicity{multiplicity};
 }
 
+Vector<int> KnotVector::DetermineMultiplicities(const Knot_* knot_vector_data,
+                                                const int knot_vector_size,
+                                                Tolerance const& tolerance) {
+  // a block to sort can come here
+
+  // create return
+  Vector<int> multiplicities;
+  multiplicities.reserve(knot_vector_size);
+
+  // initialize unique_knot - alternative is to have an iterator
+  double unique_knot{knot_vector_data[0]};
+  int multiplicity{};
+
+  // again, assumed sorted kv
+  for (int i{}; i < knot_vector_size; ++i) {
+    const Knot_& knot = knot_vector_data[i];
+    if (std::abs(knot - unique_knot) < tolerance) {
+      ++multiplicity;
+    } else {
+      multiplicities.push_back(multiplicity);
+      // not same -> update unique_knot and reset counter
+      unique_knot = knot;
+      multiplicity = 1;
+    }
+  }
+
+  return multiplicities;
+};
```

Comment:
```suggestion
}
```
Will otherwise throw a warning